### PR TITLE
Request result token (EXPOSUREAPP-10626)

### PR DESCRIPTION
**Pull request by mtwalli**
### Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10626

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingApiV1.kt [hunk 1]
```diff
@@ -20,4 +20,16 @@ interface DccTicketingApiV1 {
...
+        "X-VERSION: 1.0.0",
+        "Accept: application/jwt",
+        "Content-Type: application/json",
+    )
+    @POST
+    suspend fun getResultToken(
+        @Url url: String,
+        @Header("Authorization") authorizationHeader: String,
+        @Body body: ResultTokenRequest
+    ): Response<String>
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingApiV1.kt:34]: This won't work. Use Response<ResponseBody> and call string() of the ResponseBody to get the String.

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingApiV1.kt:34]: why it won't work ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingApiV1.kt [hunk 2]
```diff
@@ -19,5 +19,17 @@ interface DccTicketingApiV1 {
         @Url url: String,
         @Header("Authorization") authorizationHeader: String,
         @Body body: AccessTokenRequest
-    ): Response<String>
+    ): Response
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingApiV1.kt:22]: @jurajkusnier  I changed this to return okhttp response based  on @BMItr  feedback

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/common/DccJWKVerification.kt [hunk 3]
```diff
@@ -31,9 +31,10 @@ class DccJWKVerification @Inject constructor(securityProvider: SecurityProvider)
         securityProvider.setup()
     }
 
+    @Throws(DccTicketingJwtException::class)
     fun verify(jwt: String, jwkSet: Set<DccJWK>) {
         // 1. Check for empty jwkSet
-        if (jwkSet.isEmpty()) throw DccTicketingJwtException(JWT_VER_NO_JWKS)
+        if (jwkSet.isEmpty()) throw DccTicketingJwtException(JWT_VER_EMPTY_JWKS)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/common/DccJWKVerification.kt:37]: Renamed to match `specs`

Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/common/DccJWKVerification.kt:37]: **1. Check for empty jwkSet:** if `jwkSet` is empty, the operation shall abort early with error code `JWT_VER_NO_JWKS`
I don't see any `JWT_VER_EMPTY_JWKS` error code

Source: https://github.com/corona-warn-app/cwa-app-tech-spec/blob/0f1e3a323fd2d52917ffc71ea256d26278e4e16b/docs/spec/dcc-validation-service.md#verifying-the-signature-of-a-jwt-with-a-set-of-jwks

FYI check this commit: https://github.com/corona-warn-app/cwa-app-tech-spec/pull/110/commits/a24822be47637844354807e1b301513c24021323

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingApiV1.kt (hunk 2 above)
Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingApiV1.kt:22]: `retrofit2.Response` encapsulate `okhttp3.Response` together with data formatted data from the server. I already have mentioned how to use it for cert. pinning in the TODO. So I am not sure what's really improved here.
https://github.com/corona-warn-app/cwa-app-android/blob/2c2ea65709833e53e47719a2c9dd712cf61b19a3/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingServer.kt#L100

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingApiV1.kt:22]: I think what Bernd meant that if the response is `String` it won't be simply read and wrapped in Retrofit response , you have to read the response body , Did you try it ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/common/DccJWKVerification.kt (hunk 3 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/common/DccJWKVerification.kt:37]: Well the one that you linked in the ticket it does https://github.com/corona-warn-app/cwa-app-tech-spec/blob/f51d1ed62a2d40503d2f41a8e612da155fdf237b/docs/spec/dcc-validation-service.md#requesting-the-result-token

Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/common/DccJWKVerification.kt:37]: tech specs are updating frequently, better always check with the latest PR from https://github.com/corona-warn-app/cwa-app-tech-spec/pulls

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingApiV1.kt (hunk 2 above)
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/server/DccTicketingApiV1.kt:22]: I meant: Response<String> -> Response<**ResponseBody**>

**Review by jurajkusnier — APPROVED** [review]

**Review by BMItr — APPROVED** [review]

**Pull request merged**